org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenExecutionContext.java

```diff
@@ -135,21 +139,23 @@ protected MavenExecutionRequest newExecutionRequest() throws CoreException {
       request = DefaultMavenExecutionRequest.copy(parent);
     }
     if(request == null) {
-      request = createExecutionRequest();
+      request = createExecutionRequest(maven.getMavenConfiguration(), maven, maven);
+      request.setBaseDirectory(basedir);
     }
     request.setMultiModuleProjectDirectory(MavenImpl.computeMultiModuleProjectDirectory(basedir));
 
     return request;
   }
 
-  private MavenExecutionRequest createExecutionRequest() throws CoreException {
+  static MavenExecutionRequest createExecutionRequest(IMavenConfiguration mavenConfiguration,
```

Comment:
Can the same context be used wit different values for mavenConfiguraiton/lookup/maven? I was more expecting this context to be a stateful object more than a service.

Reply:
The code for creating a `MavenExecutionRequest` is also used inside the PlexusContainermanager that's the reason why I made it static here, actually we might later even move the code elsewhere as creating the request is not really bound to the execution at all.

Reply:
Is this the same code? ie should it really be factorized here? The point of IMavenExecutionContext is to be tied to the state more than being an utility class. Maybe it's one of the case where having 2 distinct methods, even with a bit a duplication, is actually better than trying to reach highest factorization possible.

Reply:
It requires the same setup, I don't think copy the code helps much here and we can do it later if we want as it is all package-private, so I think we should not complicate this too much.

Reply:
> it is all package-private

OK.